### PR TITLE
Edit item

### LINKED ISSUE #33: As a user, I want to be able to edit fields of tracked items

### PR DESCRIPTION
**Pull request by khenghun**
Enable editing of items, currently supports editing of name, quantity and description only.

### CODE REVIEW

**Review by aqcd — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/EditItemCommand.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+        return new CommandResult(String.format(MESSAGE_EDIT_ITEM_SUCCESS, editedItem));
+    }
+
+    /**
+     * Creates and returns a {@code Item} with the details of {@code itemToEdit}
+     * edited with {@code editItemDescriptor}.
+     */
+    private static Item createEditedItem(Item itemToEdit, EditItemDescriptor editItemDescriptor) {
+        assert itemToEdit != null;
```
Comment (Rahul0506) [inline src/main/java/seedu/address/logic/commands/EditItemCommand.java:96]: just use requireNonNull?

File: src/main/java/seedu/address/model/recipe/Recipe.java [hunk 2]
```diff
@@ -87,6 +87,23 @@ public Recipe delete() {
...
     }
 
+    /**
+     * Generates a printable recipe for display.
+     * @param filteredItemList The current item list which ingredients reference to.
+     * @return Printable Recipe with the same fields as this recipe.
+     */
+    public PrintableRecipe print(ObservableList<Item> filteredItemList) {
+        return new PrintableRecipe(this.id, this.ingredients, this.productId, this.productName,
+                this.productQuantity, this.description, this.isDeleted, filteredItemList);
```
Comment (Rahul0506) [inline src/main/java/seedu/address/model/recipe/Recipe.java:97]: Don't need the "this"

**Review by Rahul0506 — COMMENTED** [review]
Really small changes, optional tbh

**Pull request merged**